jdaviz/configs/specviz/plugins/line_analysis/line_analysis.py

```diff
@@ -287,14 +287,15 @@ def _uncertainty(result):
             # TODO: update specutils to allow ALL analysis to take regions and continuum so we
             # don't need these if statements
             if function == "Line Flux":
-                flux_unit = spec_subtracted.flux.unit
+                flux_unit = self.app._get_display_unit('flux')
                 if flux_unit == u.dimensionless_unscaled:
                     add_flux = True
                     flux_unit = u.Unit(self.spectrum_viewer.state.y_display_unit)
                 else:
                     add_flux = False
                 # If the flux unit is equivalent to Jy, or Jy per spaxel for Cubeviz,
                 # enforce integration in frequency space
+                flux_unit = u.Unit(flux_unit)
```

Comment:
would it be useful if `get_display_unit` had an option to cast to the unit itself?

Reply:
Yes indeed.

Reply:
added to #3112

Reply:
Thank you!

Reply:
although that is triggering a few test failures (sigh), so we may need to postpone that work, for now maybe it makes sense to just case to unit immediately so if/when we do implement that it will be easy to notice and remove the duplication

`flux_unit = u.Unit(self.app._get_display_unit('flux'))`